server/bin/pbench-verify-indexed.py

```diff
@@ -0,0 +1,187 @@
+                # NOTE: the pbench-audit-server should pick up and flag this
+                # unwanted condition.
+                continue
+            # We have a controller directory, now we scan the controller
+            # directory for all tar balls.
+            with os.scandir(c_entry.path) as controller_scan:
+                for entry in controller_scan:
+                    if not entry.is_file(follow_symlinks=False):
+                        # Tar balls can only be files.
+                        continue
+                    if entry.name.startswith("."):
+                        # Ignore any files starting with ".".
+                        continue
+                    match = tb_pat.fullmatch(entry.name)
+                    if not match:
+                        # Ignore any directory entries which do not match the
+                        # tar ball pattern.  Such entries should be flagged by
+                        # the server audit process.
+                        continue
+                    yield c_entry.name, entry.name
+
+
+def main(options):
+    if not options.cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable",
+            file=sys.stderr,
+        )
+        return 1
+
+    try:
+        config = pbench.PbenchConfig(options.cfg_name)
+    except pbench.BadConfig as e:
+        print(f"{_NAME_}: {e}", file=sys.stderr)
+        return 2
+
+    try:
+        archive_p = Path(config.ARCHIVE).resolve(strict=True)
+    except FileNotFoundError:
+        print(
+            f"The configured ARCHIVE directory, {config.ARCHIVE}, does not exist",
+            file=sys.stderr,
+        )
+        return 3
+
+    if not archive_p.is_dir():
+        print(
+            f"The configured ARCHIVE directory, {config.ARCHIVE}, is not a valid directory",
+            file=sys.stderr,
+        )
+        return 4
+
+    print("Verifying indexed state of tar balls")
+    start = pbench._time()
+
+    es = Elasticsearch(
+        [
+            {
+                "host": "elasticsearch.perf.lab.eng.bos.redhat.com",
+                "port": "9280",
+                "timeout": 600,
+            },
+        ]
+    )
+    body = {
+        "query": {
+            "filtered": {
+                "query": {"query_string": {"analyze_wildcard": True, "query": "*"}}
+            }
+        }
+    }
+    res = es.search(
+        index="dsa-pbench.v4.run.*",
+        doc_type="pbench-run",
+        expand_wildcards="open",
+        body=body,
+        fields=["@metadata.file-name"],
+        size=100000,
+    )
+    indexed = {}
+    indexed_cnt = 0
+    for hit in res["hits"]["hits"]:
+        tb_file = hit["fields"]["@metadata.file-name"][0]
+        assert tb_file not in indexed, f"Whoa, found a duplicate, f{tb_file}"
+        if tb_file.startswith("/pbench/archive"):
+            tb_file = f"/srv{tb_file}"
+        indexed[tb_file] = False
```

Comment:
This is actually a bit weird; if you find the tarball in the *index*, you're marking it "not indexed" and then if you also find it on disk you're marking it "indexed". The logic would probably be easier to follow with some renaming?

Reply:
Took a different approach.